docs/node/private-networks.mdx

```diff
@@ -0,0 +1,88 @@
+---
+sidebar_position: 7
+sidebar_label: "Private Networks"
+id: private-networks
+title: Private Networks
+description: Private Networks
+slug: /node/private-networks
+---
+
+This page describes how to set up a private network to have a granular control over who can access your network.
+
+## Private Mode
+
+Kwild nodes can be run in a private mode, where the node will only accept connections from a list of known peers. This is useful when you want to restrict access to your network to a specific set of nodes.
+
+Private mode can be enabled by setting the `private_mode` field in the `chain.p2p` section of the `config.toml` as shown below:
+
+```toml
+[chain.p2p]
+private_mode = true
+```
+
+Private mode can also be enabled by setting the `--chain.p2p.private-mode` flag when starting the node as shown below:
+
+```bash
+kwild -r /path/to/root/dir --chain.p2p.private-mode
+```
+
+
+## Whitelist Peers
+
+When running a Kwild node in private mode, it can be configured to only accept connections from a list of known peers. `whitelist_peers` field should be a comma-separated list of node IDs.
+
+Peers can be added to the whitelist by setting the `whitelist_peers` field in the `chain.p2p` section of the `config.toml` as shown below:
+
+```toml
+[chain.p2p]
+whitelist_peers = "b624b3395f22f8cb57cf86d58cbcd6b355d12e18,7b4b3b3b5f22f8cb57cf86d58cbcd6b355d12e18"
+```
+or
+
+```bash
+kwild -r /path/to/root/dir --chain.p2p.private-mode --chain.p2p.whitelist-peers="b624b3395f22f8cb57cf86d58cbcd6b355d12e18,7b4b3b3b5f22f8cb57cf86d58cbcd6b355d12e18"
+```
+
+The current set of validators, persistent peers and seed nodes are automatically added to the whitelist. Therefore, the `whitelist_peers` field is used to restrict which non-validating read-only nodes can sync data from your node's P2P endpoint.
+
+The whitelist can also be updated dynamically using the `kwil-admin peers` commands. Refer to the [Kwil Admin Peers Subcommand](/docs/ref/kwil-admin/peers) documentation for more information.
+
+:::note
+In Private mode, to establish a connection with the other node, both nodes should add each other's node ID to the whitelist. If one of the nodes doesn't add the other node to it's whitelist, the connection will not be established, unless it is a seed node or a persistent peer or a current validator.
+:::
```

Comment:
```suggestion
Seed nodes and persistent peers are automatically whitelisted. In Private mode, if the existing node is not a persistent peer or seed node, the connecting node should also add the existing node's ID to its whitelist. 
:::
```

I think something like this, just to make the point immediately that mutual whitelisting is only required if the connected node is not a persistent peer or a seed node.

Reply:
Wait so thinking about this practically, when would a connecting node need to add an existing node to its whitelist? How else can it connect without persistent peer or seed nodes?

Reply:
none of the persistent peers or seed nodes needs to be whitelisted, but other nodes which are not listed as seeds or persistent peers need to be whitelisted for establishing a mesh connectivity. 

consider 3 nodes: n1 <-> n2, n3(new node)
 suppose n3 is joining the network through n2, here n3 is indirectly connected to n1. So if n1 receives any packet, it broadcasts it to n2 which eventually broadcasts to n3 (takes 2 hops for the packets to reach n3). If n2 goes down, the network breaks. So its good to have 1-1 connectivity with a lot more nodes.  In that case u would add those to your whitelist. 

Cometbft also has a limit on number of persistent peers u can have for better performance, so rest of the nodes u r still connected to, but cometbft doesnt ensure persistence of connection with these nodes.

Reply:
Got it, that makes sense. Thank you for explaining!

I would propose changing it to something like this then:

```suggestion
Seed nodes and persistent peers are automatically whitelisted. In order for a new node to maintain a connection with an existing node that is not a seed or persistent peer, the existing node must also add the new node to its whitelist.
:::
```